src/unix/core.c

```diff
@@ -1285,3 +1292,33 @@ int uv_os_unsetenv(const char* name) {
 
   return 0;
 }
+
+
+int uv_os_gethostname(char* buffer, size_t* size) {
+  /*
+    On some platforms, if the input buffer is not large enough, gethostname()
+    succeeds, but truncates the result. libuv can detect this and return ENOBUFS
+    instead by creating a large enough buffer and comparing the hostname length
+    to the size input.
+  */
+  char buf[MAXHOSTNAMELEN + 1];
+  size_t len;
+
+  if (buffer == NULL || size == NULL || *size == 0)
```

Comment:
Its not possible to call with a NULL buffer, and get the required size? Its a useful pattern, but consistency with other similar uv functions is more important.